.github/workflows/e2e.yaml

```diff
@@ -116,21 +116,154 @@ jobs:
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-qemu-action@v2
+      - uses: docker/setup-buildx-action@v2
+        with:
+          driver-opts: network=host
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.20"
+          cache: true
+      - name: Install Ngrok Tunnel
+        run: |
+          curl -s https://ngrok-agent.s3.amazonaws.com/ngrok.asc | sudo tee /etc/apt/trusted.gpg.d/ngrok.asc >/dev/null && \
+          echo "deb https://ngrok-agent.s3.amazonaws.com buster main" | sudo tee /etc/apt/sources.list.d/ngrok.list && \
+          sudo apt update && \
+          sudo apt install ngrok
+          touch ~/ngrok.log
+          (timeout 30m ngrok http 80 --authtoken ${{ secrets.NGROK_AUTHTOKEN }} --log ~/ngrok.log) &
+          echo $?
+      - name: Install kind
+        run: |
+          curl -sSLo kind "https://github.com/kubernetes-sigs/kind/releases/download/v0.19.0/kind-linux-amd64"
+          chmod +x kind
+          sudo mv kind /usr/local/bin/kind
+          kind version
+      - name: Install Kubectl
+        run: |
+          curl -sSLO "https://storage.googleapis.com/kubernetes-release/release/v1.26.1/bin/linux/amd64/kubectl"
+          chmod +x kubectl
+          sudo mv kubectl /usr/local/bin/kubectl
+          kubectl version --client --output=yaml
+      - name: Kubernetes KinD Cluster
+        run: |
+          make init-kind
+      - name: install nginx
+        run: |
+          make init-nginx
+      - name: install workflows
+        run: |
+          make init-argo-workflows
+      - name: adding license env var
+        run: |
+          echo "GITLAB_LICENSE=${{ secrets.GITLABLICENSE }}" >> $GITHUB_ENV
+      - name: install gitlab
+        run: |
+          echo "GITLAB_LICENSE is $GITLAB_LICENSE"  # Debugging step
+          make init-gitlab
+      - name: populate gitlab
+        run: |
+          output=$(sh ./scripts/setup-gitlab.sh | tail -n1)
+          echo "GITLAB_TOKEN=$(echo $output | cut -d' ' -f1)" >> $GITHUB_ENV
+      - name: Build Docker Image
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
+          tags: localhost:5001/piper:latest
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Check tunnel existence
+        run: |
+          echo "NGROK_URL=$(cat ~/ngrok.log | grep -o 'url=https://.*' | cut -d '=' -f 2)" >> $GITHUB_ENV
+          cat ~/ngrok.log | grep -o 'url=https://.*' | cut -d '=' -f 2
+      - name: init piper
+        run: |
+          helm upgrade --install piper ./helm-chart \
+          -f ./examples/template.values.dev.yaml \
+          --set piper.gitProvider.name="gitlab" \
+          --set piper.gitProvider.token="${{ env.GITLAB_TOKEN }}" \
+          --set piper.gitProvider.url="http://gitlab-webservice-default.gitlab:8080" \
+          --set piper.gitProvider.webhook.url="${{ env.NGROK_URL }}/piper/webhook" \
```

Comment:
It is used in same cluster, we don't need use Ngrok at all .. 
Just set the "piper_service_name.default/webhook" as the address